src/_langs/en/updates/2015-07-22-access-ble-devices-with-web-bluetooth.markdown

```diff
@@ -0,0 +1,279 @@
+  console.log(device.name);
+  // Indicates whether or not the device is paired with the system.
+  console.log(device.paired);
+  // UUIDs of protocols, profiles and services advertised by the device.
+  console.log(device.uuids);
+
+  // Attempts to connect to GATT Remote Server.
+  return device.connectGATT();
+})
+.then(server => {...})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+
+### Read a Bluetooth Characteristic
+
+Here we are connected to the GATT Remote Server of the Bluetooth device. Now we
+want to get a Primary Bluetooth GATT Service and read a characteristic that
+belongs to this service. Let's try for instance to read the current charge
+level of a connected device battery.
+
+In the example below, `battery_level` is the [adopted Bluetooth GATT
+Characteristic Battery
+Level](https://developer.bluetooth.org/gatt/characteristics/Pages/CharacteristicViewer.aspx?u=org.bluetooth.characteristic.battery_level.xml).
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => device.connectGATT())
+.then(server => {
+  // Getting Battery Service...
+  return server.getPrimaryService('battery_service');
+})
+.then(service => {
+  if (!service) throw 'Battery Service not found';
+  // Getting Battery Level Characteristic...
+  return service.getCharacteristic('battery_level');
+})
+.then(characteristic => {
+  // Reading Battery Level...
+  return characteristic.readValue();
+})
+.then(buffer => {
+  var data = new DataView(buffer);
+  console.log('Battery percentage is ' + data.getUint8(0));
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+If you use a custom Bluetooth GATT characteristic,
+`window.BluetoothUUID.getCharacteristic` definitely comes in handy. Calling it
+with `0x5678` will return `"00005678-0000-1000-8000-00805f9b34fb"`.
+
+{% highlight javascript %}
+...
+.then(service => {
+  var myCharacteristic = BluetoothUUID.getCharacteristic(0x5678);
+  return service.getCharacteristic(myCharacteristic);
+})
+{% endhighlight %}
+
+### Write to a Bluetooth Characteristic
+
+Writing to a Bluetooth GATT Characteristic is as easy as reading it. This time,
+let's reset the value of the Energy Expended field in the Heart Rate
+Measurement characteristic to 0 on a heart rate monitor device.
+
+There is no magic there I promise. It's all explained in the [Heart Rate
+Control Point Characteristic
+page](https://developer.bluetooth.org/gatt/characteristics/Pages/CharacteristicViewer.aspx?u=org.bluetooth.characteristic.heart_rate_control_point.xml).
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+.then(device => device.connectGATT())
+.then(server => server.getPrimaryService('heart_rate'))
+.then(service => service.getCharacteristic('heart_rate_control_point'))
+.then(characteristic => {
+  var resetEnergyExpended = new Uint8Array([1]);
```

Comment:
Comment here to briefly explain that writing 1 is the signal to reset energy expended.